java/src/Ice/src/main/java/com/zeroc/Ice/UnknownException.java

```diff
@@ -1,42 +1,24 @@
-//
-// Copyright (c) ZeroC, Inc. All rights reserved.
-//
-//
-// Ice version 3.7.10
-//
-// <auto-generated>
-//
-// Generated from file `LocalException.ice'
-//
-// Warning: do not edit this file.
-//
-// </auto-generated>
-//
+// Copyright (c) ZeroC, Inc.
 
 package com.zeroc.Ice;
 
 /**
- * This exception is raised if an operation call on a server raises an unknown exception. For
- * example, for C++, this exception is raised if the server throws a C++ exception that is not
- * directly or indirectly derived from <code>Ice::LocalException</code> or <code>Ice::UserException
- * </code>.
+ * This exception is raised if an operation call on a server raises an unknown exception, that is,
+ * any exception which does not derive from {@link LocalException} or {@link UserException}.
  */
 public class UnknownException extends LocalException {
-  public UnknownException() {
-    this.unknown = "";
-  }
-
   public UnknownException(Throwable cause) {
     super(cause);
     this.unknown = "";
   }
 
   public UnknownException(String unknown) {
+    super("unknown exception with type ID '" + unknown + "'");
     this.unknown = unknown;
   }
 
   public UnknownException(String unknown, Throwable cause) {
-    super(cause);
+    super("unknown exception with type ID '" + unknown + "'", cause);
```

Comment:
This is bogus.

Reply:
Fixed, after discussing what this parameter actually holds.